### PR TITLE
[Docs] Filter Cog Guide

### PR DESCRIPTION
**Pull request by bobloy**
### Type

- [ ] Bugfix
- [X] Enhancement
- [ ] New feature

### Description of the changes

User guide for the Filter cog. Docs were automatically generated using the same method as #4488 
A step towards the #1734 goal

### CODE REVIEW

**Comment by bobloy** [comment]
This is now ready for review. If #4599 is accepted, will fix `[words...]` to `<words...>`

File: docs/cog_guides/filter.rst
Comment (Flame442) [inline docs/cog_guides/filter.rst]:
```suggestion
This cog is designed for "filtering" unwanted words and phrases from a server.
```
Everywhere else in this document, "server" is used, so either consistently use "guild" or change this instance.

File: docs/cog_guides/filter.rst
Comment (Flame442) [inline docs/cog_guides/filter.rst]:
```suggestion
Base command to add or remove words from the server filter.
```
I'm also not a fan of "*add* or *remove* words *from*", but there isn't a good way to make this sound better besides being overly verbose ("add words to or remove words from"). Unless you can think of a better wording I'm fine leaving that how it is.

File: docs/cog_guides/filter.rst
Comment (Flame442) [inline docs/cog_guides/filter.rst]:
```suggestion
Base command to add or remove words from the channel filter.
```

File: redbot/cogs/filter/filter.py
Comment (Flame442) [inline redbot/cogs/filter/filter.py]:
```suggestion
    """This cog is designed for "filtering" unwanted words and phrases from a server.
```

File: redbot/cogs/filter/filter.py
Comment (Flame442) [inline redbot/cogs/filter/filter.py]:
```suggestion
        """Base command to add or remove words from the server filter.
```

File: redbot/cogs/filter/filter.py
Comment (Flame442) [inline redbot/cogs/filter/filter.py]:
```suggestion
        """Base command to add or remove words from the channel filter.
```

**Review by Flame442 — CHANGES_REQUESTED** [review]
Some minor consistency and grammar issues, but otherwise solid. Thanks for contributing.

**Review by Flame442 — DISMISSED** [review]
Good to go once the merge conflict is resolved!

**Review by Flame442 — APPROVED** [review]
📦

**Pull request merged**
